fuel/datasets/text.py

```diff
@@ -75,20 +91,27 @@ class TextFile(Dataset):
 
     def __init__(self, files, dictionary, bos_token='<S>', eos_token='</S>',
                  unk_token='<UNK>', level='word', preprocess=None,
-                 encoding=None):
+                 encoding=None, contains_unk=True):
         self.files = files
         self.dictionary = dictionary
         if bos_token is not None and bos_token not in dictionary:
-            raise ValueError
+            raise ValueError(
+                "BOS token '{}' is not in the dictionary".format(bos_token))
         self.bos_token = bos_token
         if eos_token is not None and eos_token not in dictionary:
-            raise ValueError
+            raise ValueError(
+                "EOS token '{}' is not in the dictionary".format(eos_token))
         self.eos_token = eos_token
         if unk_token not in dictionary:
```

Comment:
Why not just do the same for the unknown-token as for the others i.e. allow `unk_token=None` to be passed and check `if unk_token is not None and unk_token not in dictionary`. Saves an extra keyword argument, plus it's more consistent.